python/lsst/ip/isr/vignette.py

```diff
@@ -52,39 +53,136 @@ class VignetteConfig(Config):
         doc="Number of points used to define the vignette polygon.",
         default=100,
     )
-    doWriteVignettePolygon = Field(
-        dtype=bool,
-        doc="Persist polygon used to define vignetted region?",
-        default=False,
-    )
 
 
 class VignetteTask(Task):
-    """Define a simple circular vignette pattern.
+    """Define a simple circular vignette pattern and optionally update mask
+    plane.
     """
     ConfigClass = VignetteConfig
     _DefaultName = "isrVignette"
 
-    def run(self, exposure):
+    def run(self, exposure=None, doUpdateMask=True, maskPlane="NO_DATA", vignetteValue=None, log=None):
         """Generate circular vignette pattern.
 
         Parameters
         ----------
         exposure : `lsst.afw.image.Exposure`
```

Comment:
This `exposure` parameter seems now optional, so it should be indicated as such in the doc string.  In addition, the new parameters should also be documented in the docstring.